### PR TITLE
[json-rpc-cxx] Add new port

### PR DESCRIPTION
**Pull request by AenBleidd**

### CODE REVIEW

File: ports/json-rpc-cxx/fix-config.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ add_library(json-rpc-cxx INTERFACE)
+-target_include_directories(json-rpc-cxx INTERFACE include)
++target_include_directories(json-rpc-cxx INTERFACE
++        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}>
++        $<INSTALL_INTERFACE:include>
++        )
+ install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
++install (TARGETS json-rpc-cxx EXPORT json-rpc-cxx-targets)
++install (EXPORT json-rpc-cxx-targets
++        FILE json-rpc-cxx-config.cmake
```
Comment (MonicaLiu0311) [inline ports/json-rpc-cxx/fix-config.patch:17]:
```suggestion
+        FILE  unofficial-json-rpc-cxx-config.cmake
```

File: ports/json-rpc-cxx/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DCOMPILE_TESTS=OFF
+        -DCOMPILE_EXAMPLES=OFF
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
+file(READ "${CURRENT_PACKAGES_DIR}/share/${PORT}/${PORT}-config.cmake" JSON_RPC_CXX_CONFIG)
+file(WRITE "${CURRENT_PACKAGES_DIR}/share/${PORT}/${PORT}-config.cmake" "
```
Comment (MonicaLiu0311) [inline ports/json-rpc-cxx/portfile.cmake:23]:
```suggestion
file(READ "${CURRENT_PACKAGES_DIR}/share/${PORT}/ unofficial-${PORT}-config.cmake" JSON_RPC_CXX_CONFIG)
file(WRITE "${CURRENT_PACKAGES_DIR}/share/${PORT}/ unofficial-${PORT}-config.cmake" "

```

**Review by MonicaLiu0311 — DISMISSED** [review]

**Comment by AenBleidd** [comment]
@MonicaLiu0311, if I apply suggested changes, then I can't use this port:

```
CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
  Could not find a package configuration file provided by "json-rpc-cxx" with
  any of the following names:

    json-rpc-cxxConfig.cmake
    json-rpc-cxx-config.cmake

  Add the installation prefix of "json-rpc-cxx" to CMAKE_PREFIX_PATH or set
  "json-rpc-cxx_DIR" to a directory containing one of the above files.  If
  "json-rpc-cxx" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  CMakeLists.txt:5 (find_package)
```

**Review by MonicaLiu0311 — CHANGES_REQUESTED** [review]

**Comment by MonicaLiu0311** [comment]
> @MonicaLiu0311, if I apply suggested changes, then I can't use this port:
> 
> ```
> CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
>   Could not find a package configuration file provided by "json-rpc-cxx" with
>   any of the following names:
> 
>     json-rpc-cxxConfig.cmake
>     json-rpc-cxx-config.cmake
> 
>   Add the installation prefix of "json-rpc-cxx" to CMAKE_PREFIX_PATH or set
>   "json-rpc-cxx_DIR" to a directory containing one of the above files.  If
>   "json-rpc-cxx" provides a separate development package or SDK, be sure it
>   has been installed.
> Call Stack (most recent call first):
>   CMakeLists.txt:5 (find_package)
> ```

Also modify the usage of the response.

**Comment by MonicaLiu0311** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review".

File: ports/json-rpc-cxx/portfile.cmake (hunk 2 above)
Comment (AenBleidd) [reply ports/json-rpc-cxx/portfile.cmake]: Fixed

File: ports/json-rpc-cxx/fix-config.patch [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
++target_include_directories(json-rpc-cxx INTERFACE
++        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}>
++        $<INSTALL_INTERFACE:include>
++        )
+ install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
++install (TARGETS json-rpc-cxx EXPORT json-rpc-cxx-config)
++install (EXPORT json-rpc-cxx-config
++        NAMESPACE unofficial::json-rpc-cxx::
++        DESTINATION share/json-rpc-cxx
++        )
```
[the remark this answers is not in the document]
Comment (AenBleidd) [reply ports/json-rpc-cxx/fix-config.patch]: Fixed

**Comment by AenBleidd** [comment]
@MonicaLiu0311, fixed, and ready for review

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
json-rpc-cxx provides CMake targets:

    find_package(unofficial-json-rpc-cxx CONFIG REQUIRED)
    target_link_libraries(main PRIVATE unofficial::json-rpc-cxx::json-rpc-cxx)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**